ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/MaxResourceIntensivePerBatchJsonRpcHttpServiceTest.java

```diff
@@ -0,0 +1,125 @@
+      final JsonArray json = parseResponse(resp);
+
+      // Assert that the first non-resource-intensive request executed successfully (it should not
+      // contain an error).
+      assertThat(json.getJsonObject(0).getString("error")).isNull();
+
+      // Assert that the first resource-intensive request executed successfully.
+      // As per our configuration, only one resource-intensive request should be allowed per batch,
+      // so it should not exceed the limit.
+      assertThat(json.getJsonObject(1).getJsonObject("error").getInteger("code"))
+          .isNotEqualTo(JsonRpcError.EXCEEDS_RPC_MAX_RESOURCE_INTENSIVE_BATCH_SIZE.getCode());
+
+      // Assert that the second non-resource-intensive request executed successfully (it should not
+      // contain an error).
+      assertThat(json.getJsonObject(2).getString("error")).isNull();
+
+      // Assert that the second resource-intensive request failed due to exceeding the limit of
+      // resource-intensive requests per batch.
+      assertThat(json.getJsonObject(3).getJsonObject("error").getInteger("code"))
+          .isEqualTo(JsonRpcError.EXCEEDS_RPC_MAX_RESOURCE_INTENSIVE_BATCH_SIZE.getCode());
+
+      // Assert that the third non-resource-intensive request executed successfully (it should not
+      // contain an error).
+      assertThat(json.getJsonObject(4).getString("error")).isNull();
+    }
+  }
+
+  @Test
+  public void shouldOnlyExecuteOneResourceRequestOPerBatchWhenConfigIsDisabled() throws Exception {
+    // disable resource-intensive methods per batch
+    final int resourceIntensiveLimit = -1;
+    initMaxResourceIntensivePerBatchSize(resourceIntensiveLimit);
+
+    final RequestBody body = createRequestBody();
+
+    // Execute the batch request and parse the response as a JSON array.
+    try (final Response resp = client.newCall(buildPostRequest(body)).execute()) {
+      final JsonArray json = parseResponse(resp);
+
+      // Iterate over the responses. Since we've disabled the limit on resource-intensive methods
+      // per batch,
+      // none of the requests should fail due to exceeding the batch size limit (i.e., no error
+      // should have the code EXCEEDS_RPC_MAX_BATCH_SIZE).
+      for (int i = 0; i < json.size(); i++) {
+        JsonObject jsonObject = json.getJsonObject(i);
+        if (jsonObject.containsKey("error")) {
+          assertThat(jsonObject.getJsonObject("error").getInteger("code"))
+              .isNotEqualTo(JsonRpcError.EXCEEDS_RPC_MAX_RESOURCE_INTENSIVE_BATCH_SIZE.getCode());
+        }
+      }
+    }
+  }
+
+  // Create a batch request containing two types of requests: one resource-intensive and one
+  // non-resource-intensive.
+  // The batch request is structured as a JSON array containing five requests in total.
+  private RequestBody createRequestBody() {
+    return RequestBody.create(
+        "["
+            // 1 - Non-resource-intensive request
```

Comment:
I don't think we need these comments 1-5 - the method name is a good explanation